optimizer_wrapper.rb

```diff
@@ -363,78 +374,84 @@ def self.split_independent_vrp_by_skills(vrp)
         independent_skills[set_index] += b_set
       }
     }
-
-    # Original skills are retrieved
-    independant_skill_sets = independent_skills.map{ |index_set|
+    # Independent skill sets : Original skills are retrieved
+    independent_skills.map{ |index_set|
       index_set.collect{ |index| mission_skills[index] }
     }
+  end
 
-    unused_vehicles_indices = (0..vrp.vehicles.size - 1).to_a
-    independent_vrps = independant_skill_sets.collect{ |skills_set|
+  def self.build_independent_vrps(vrp, skill_sets, unused_vehicle_indices, skill_vehicle_ids, skill_service_ids)
+    independent_vrps = skill_sets.collect{ |skills_set|
       # Compatible problem ids are retrieved
-      vehicles_indices = skills_set.flat_map{ |skills| skill_vehicle_ids.select{ |k, _v| (k & skills) == skills }.flat_map{ |_k, v| v } }.uniq
-      vehicles_indices.each{ |index| unused_vehicles_indices.delete(index) }
+      vehicle_indices = skills_set.flat_map{ |skills|
+        skill_vehicle_ids.select{ |k, _v| (k & skills) == skills }.flat_map{ |_k, v| v }
+      }.uniq
+      vehicle_indices.each{ |index| unused_vehicle_indices.delete(index) }
       service_ids = skills_set.flat_map{ |skills| skill_service_ids[skills] }
```

Comment:
I am not sure _skill_service_ids_ keys correspond to _skill_sets_ after calling _compute_independent_skills_sets_. Are they?

Reply:
Could you maybe add a test with trip relation and different skills on each vehicle_trip to check this ?

Reply:
_skill_service_ids_ is untouched by _compute_independent_skills_sets_. It only give the service ids associated to a given skill set.